monasca_events_api/tests/test_stream_definitions.py

```diff
@@ -0,0 +1,518 @@
+    def test_on_post_integrity_error(
+            self,
+            validate,
+            tenantid,
+            json,
+            readjson,
+            streamvalid,
+            getname,
+            desc,
+            fireactions,
+            expire,
+            repo):
+        """POST method  failed due to integrity error"""
+        validate.return_value = True
+        repo.connect.side_effect = AlreadyExistsException()
+        fireactions.return_value = "fire_actions"
+        getname.return_value = "Test"
+        expire.return_value = "expire_actions"
+        desc.return_value = "Stream_Description"
+        readjson.return_value = {
+            u'fire_criteria': [
+                {
+                    u'event_type': u'compute.instance.create.start'},
+                {
+                    u'event_type': u'compute.instance.create.end'}],
+            u'description': u'provisioning duration',
+            u'group_by': [u'instance_id'],
+            u'expiration': 90000,
+            u'select': [
+                {
+                    u'event_type': u'compute.instance.create.*'}],
+            u'name': u'buzz'}
+
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        streamvalid.return_value = True
+
+        streamsObj = StreamDefinitionsSubClass()
+        streamsObj._stream_definitions_repo = StreamsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            streamsObj.on_post(self._generate_req(), res)
+            self.assertFalse(
+                1,
+                msg="DB Integrity Error, should fail but passed")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPConflict)
+            self.assertEqual(e.status, '409 Conflict')
+
+    @mock.patch('monasca_events_api.v2.common.helpers.add_links_to_resource')
+    @mock.patch(
+        'monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.StreamDefinitions._stream_definition_create')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_expire_actions')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_fire_actions')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_description')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.get_query_stream_definition_name')
+    @mock.patch(
+        'monasca_events_api.v2.stream_definitions.StreamDefinitions._validate_stream_definition')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_json_msg_body')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    def test_on_post_pass__validate_stream_definition(
+            self,
```

Comment:
Same questions on this and the on_post_pass test as I had on the test_events version of them.